js-api-doc/logger/index.d.ts

```diff
@@ -51,6 +54,7 @@ export interface Logger {
     message: string,
     options: {
       deprecation: boolean;
+      deprecationType?: Deprecation;
```

Comment:
We should probably type this explicitly so that `deprecationType` is set if and only if `deprecation` is true.

Reply:
I think this is the best option